ui/app/pages/send/send-content/add-recipient/ens-input.component.js

```diff
@@ -53,20 +54,30 @@ export default class EnsInput extends Component {
     }
   }
 
-  // If an address is sent without a nickname, meaning not from ENS or from
-  // the user's own accounts, a default of a one-space string is used.
   componentDidUpdate (prevProps) {
     const {
       input,
     } = this.state
     const {
       network,
+      value,
     } = this.props
 
+    let newValue
+
+    // Set the value of our input based on QR code provided by parent
+    if (input !== value && prevProps.value !== value) {
+      newValue = value
+    }
+
     if (prevProps.network !== network) {
       const provider = global.ethereumProvider
       this.ens = new ENS({ provider, network })
-      this.onChange({ target: { value: input } })
+      newValue = input
```

Comment:
This unfortunately will still erase the address from the QR code scanner if it's passed in on the same render as the network change.